### ISSUE TITLE
Observation/Question: What the heck is happening with the bounding box?

### ISSUE THREAD

**Issue by dancesWithCycles** [derhuerst/hafas-monitor-trips#12]
Hi folks,
I am calling this instruction set.

```
npm i
npm run build
./start
```

This is the response.

```
$ export BBOX='{"north": 54.52, "west": 6.54, "south": 51.29, "east": 6.65}'
$ echo $BBOX
{"north": 54.52, "west": 6.54, "south": 51.29, "east": 6.65}
$ ./start.sh
{"level":30,"time":1707769699513,"pid":2319214,"name":"serve","msg":"listening on port 3000"}
...
{"level":30,"time":1707769699783,"pid":2319212,"name":"monitor","msg":"serving Prometheus metrics on port 9323"}
bbox: {"north":53.0665,"west":6.5895,"south":52.7435,"east":6.6005}
bbox: {"north":53.228,"west":6.584,"south":52.582,"east":6.606}
bbox: {"north":53.551,"west":6.573,"south":52.259,"east":6.617}
bbox: {"north":54.197,"west":6.551,"south":51.613,"east":6.639}
bbox: {"north":55.489,"west":6.507,"south":50.321,"east":6.683}
bbox: {"north":58.073,"west":6.419,"south":47.737,"east":6.771}
bbox: {"north":63.241,"west":6.243,"south":42.569,"east":6.947}
bbox: {"north":73.577,"west":5.891,"south":32.233,"east":7.299}
{"level":20,"time":1707769700724,"pid":2319213,"name":"match","origTripId":"1|13666|0|86|12022024","isMatched":false,"matchTime":2,"waitTime":1,"msg":"failed to match fresh movement in 2 after 1 waiting"}
...
bbox: {"north":85.751,"west":5.1872,"south":11.561,"east":8.0028}
{"level":20,"time":1707769700837,"pid":2319213,"name":"match","origTripId":"1|66042|0|86|12022024","isMatched":false,"matchTime":0,"waitTime":1,"msg":"failed to match fresh movement in 0 after 1 waiting"}
...
bbox: {"north":44.407,"west":3.7804,"south":-29.783,"east":9.4096}
{"level":20,"time":1707769701001,"pid":2319213,"name":"match","origTripId":"1|66042|0|86|12022024","isMatched":false,"matchTime":2,"waitTime":0,"msg":"failed to match fresh movement in 2 after 0 waiting"}
...
bbox: {"north":-38.281,"west":0.9745,"south":-67.529,"east":12.2155}
bbox: {"north":23.657,"west":-4.5783,"south":82.153,"east":17.7683}
{"level":50,"time":1707769701218,"pid":2319212,"name":"monitor","err":{"type":"Error","message":"north (23.657) must be larger than south (82.153).","stack":"Error: north (23.657) must be larger than south (82.153).\n    at Object.radar (/home/pacs/sib00/users/gtfsr_srv_fp/git/berlin-gtfs-rt-server/node_modules/hafas-client/index.js:604:29)\n    at atomWithCache (/home/pacs/sib00/users/gtfsr_srv_fp/git/berlin-gtfs-rt-server/node_modules/cached-hafas-client/index.js:193:38)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async findMaxRadarResults (/home/pacs/sib00/users/gtfsr_srv_fp/git/berlin-gtfs-rt-server/node_modules/hafas-monitor-trips/lib/find-max-radar-results.js:57:21)\n    at async cachedFindMaxRadarResults (/home/pacs/sib00/users/gtfsr_srv_fp/git/berlin-gtfs-rt-server/node_modules/hafas-monitor-trips/lib/find-max-radar-results.js:90:21)"},"msg":"north (23.657) must be larger than south (82.153)."}
...
```

It looks like the bounding box I exported `export BBOX='{"north": 54.52, "west": 6.54, "south": 51.29, "east": 6.65}'` is altered until it is not valid anymore.

I added some extra logs to the console for debugging purposes.

$ vi .../git/berlin-gtfs-rt-server/node_modules/hafas-client/index.js
```
...
        if ('number' !== typeof east) throw new TypeError('east must be a number.')
        if (north <= south) throw new Error('north (' + north + ') must be larger than south (' + south + ').')//TODO  2024-02-12
        if (east <= west) throw new Error('east must be larger than west.')
...
```

$ vi .../git/berlin-gtfs-rt-server/node_modules/hafas-monitor-trips/lib/find-max-radar-results.js
```
...
        const t0 = Date.now()
        console.log('bbox: ' + JSON.stringify(bbox));//TODO begerad 2024-02-12
        const movements = await hafas.radar(bbox, {
...
```

However, I do not observe this behavior when I am using the bounding box from the [readme](https://github.com/derhuerst/berlin-gtfs-rt-server/blob/7e9e095878ae3ea13c976fee1fa3e39643ed28b2/readme.md?plain=1#L57).

```
...
$ export BBOX='{"north": 52.52, "west": 13.36, "south": 52.5, "east": 13.39}'
$ echo $BBOX
{"north": 52.52, "west": 13.36, "south": 52.5, "east": 13.39}
sib00-gtfsr_srv_fp@h48:~/git/berlin-gtfs-rt-server$ ./start.sh
{"level":30,"time":1707770463521,"pid":2323054,"name":"monitor","msg":"serving Prometheus metrics on port 9323"}
{"level":30,"time":1707770463527,"pid":2323056,"name":"serve","msg":"listening on port 3000"}
{"level":30,"time":1707770463528,"pid":2323056,"name":"serve","msg":"serving Prometheus metrics on port 9325"}
{"level":30,"time":1707770463546,"pid":2323055,"name":"match","msg":"serving Prometheus metrics on port 9324"}
{"level":20,"time":1707770463918,"pid":2323055,"name":"match","origTripId":"1|46423|22|86|12022024","isMatched":false,"matchTime":2,"waitTime":1,"msg":"failed
...
```

Any idea what causes this observation?

Cheers!

**Comment by derhuerst** [comment]
Although the bounding box you defined is *very* unusual (a 7x360km north-south strip, from somewhere in the North Sea to Düsseldorf, through Netherlands), it should definitely work.

**Comment by dancesWithCycles** [comment]
That is about the area the [Fahrplaner](https://fahrplaner) is serving. It includes areas in HB, NI, HH, SH and MV.

**Comment by derhuerst** [comment]
Are you sure?

<img width="498" alt="Screenshot 2024-02-21 at 22 03 40" src="https://github.com/derhuerst/hafas-monitor-trips/assets/5072613/85fb14bb-1d80-4ad6-8eb2-29f1250dc81d">

**Comment by derhuerst** [comment]
While `hafas-monitor-trips`'s current bounding box calculation logic computes ridiculously large bounding boxes with your example input, the actual problem was that `hafas-client` assumed & asserted that `bbox.north <= bbox.south`.

I have fixed this in https://github.com/public-transport/hafas-client/commit/90b114040163e13a69840707818985fec81e9a39 and published the fix as `hafas-client@6.2.2`, so an `npm upgrade` should make the error message disappear.
